jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,735 @@
+        {
+            return _pattern.test(getInetAddress(request.getConnectionMetaData().getRemoteSocketAddress()));
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return _pattern.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            return obj instanceof InetAddressPatternPredicate inetAddressPatternPredicate && _pattern.equals(inetAddressPatternPredicate._pattern);
+        }
+
+        @Override
+        public String toString()
+        {
+            return super.toString();
+        }
+    }
+
+    /**
+     * A {@link Predicate} over {@link Request} that tests {@link Request#getMethod() method} name.
+     * Using predicates in less efficient than using {@link ConditionalHandler#includeMethod(String...)}
+     * and {@link ConditionalHandler#excludeMethod(String...)}, so this predicate should only be used
+     * if necessary to combine with other predicates.
+     */
+    public static class MethodPredicate implements Predicate<Request>
+    {
+        private final String _method;
+
+        public MethodPredicate(String method)
+        {
+            _method = Objects.requireNonNull(method);
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _method.equals(request.getMethod());
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s@%x{%s}", getClass().getSimpleName(), hashCode(), _method);
+        }
+    }
+
+    /**
+     * A {@link Predicate} over {@link Request} that tests a {@link PathSpec} against
+     * the {@link Request#getPathInContext(Request) pathInContext}.
+     * Using predicates in less efficient than using {@link ConditionalHandler#include(PathSpec...)}
+     * and {@link ConditionalHandler#exclude(PathSpec...)}, so this predicate should only be used
+     * if necessary to combine with other predicates.
+     *
+     */
+    public static class PathSpecPredicate implements Predicate<Request>
+    {
+        private final PathSpec _pathSpec;
+
+        public PathSpecPredicate(PathSpec pathSpec)
+        {
+            _pathSpec = Objects.requireNonNull(pathSpec);
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _pathSpec.matches(Request.getPathInContext(request));
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s@%x{%s}", getClass().getSimpleName(), hashCode(), _pathSpec);
+        }
+    }
+
+    /**
+     * An implementation of {@link ConditionalHandler} that will invoke {@link #nextHandle(Request, Response, Callback)} if
+     * the conditions are met, otherwise {@code false} is returned from {@link #doNotHandle(Request, Response, Callback)} and
+     * no further handling is done.
+     */
+    public static class Terminate extends ConditionalHandler
```

Comment:
I'm OK with `DontHandle`.

But I don't understand what you say after that?  A class should always be named after what it does.     The whole point of `DontHandle`, `Forbidden` and `SkipNext` are that they are utility concrete classes that can be used directly rather than extending Conditional handler.

If you really don't like them, then we can remove `DontHandle` and `Forbidden`.  We need to keep `SkipNext` as that is the one we can tell users to use in XML that will make any handler conditional.

Reply:
Ah I think I now understand after @janbartel said the same thing a different way.  See the new names/javadoc and inverted logic.